From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: TheDBP <TheDBP@users.noreply.github.com>
Date: Sat, 19 Sep 2026 20:44:45 -0400
Subject: [PATCH] bonito: replace the HIDL lights stack with the LineageOS AIDL
 lights HAL

lineage-24.0 deleted hardware/interfaces/light/2.0 and
hardware/google/interfaces/light/1.0, taking hardware.google.light@1.0-service
(the wrapper around the legacy lights.qcom module) with them, and
LightsService is AIDL-only now.

android.hardware.light-service.lineage drives the same sysfs nodes
lights.c did -- /sys/class/leds/{red,green,blue} with blink,
/sys/class/leds/button-backlight, /sys/class/backlight/panel0-backlight
-- all labelled sysfs_leds by genfs_contexts, which system policy already
grants hal_light. Screen brightness goes through the sdm845 composer's
SetDisplayBrightness regardless. Dropped: the hardware.google.light
extension (high-brightness mode / LED calibration) that only the Google
wrapper exposed.
---
 device.mk                                      |  3 +--
 device_framework_matrix.xml                    |  8 --------
 manifest.xml                                   | 18 ------------------
 sepolicy/vendor/qcom/common/file_contexts      |  2 +-
 sepolicy/vendor/qcom/common/hwservice_contexts |  1 -
 5 files changed, 2 insertions(+), 30 deletions(-)

diff --git a/device.mk b/device.mk
index 27a44270..fefd5c93 100644
--- a/device.mk
+++ b/device.mk
@@ -340,8 +340,7 @@ PRODUCT_PACKAGES += \
 
 # Light HAL
 PRODUCT_PACKAGES += \
-    lights.qcom \
-    hardware.google.light@1.0-service
+    android.hardware.light-service.lineage
 
 PRODUCT_PROPERTY_OVERRIDES += \
     ro.hardware.lights=qcom
diff --git a/device_framework_matrix.xml b/device_framework_matrix.xml
index 71b388e6..de81be3d 100644
--- a/device_framework_matrix.xml
+++ b/device_framework_matrix.xml
@@ -149,14 +149,6 @@
             <instance>default</instance>
         </interface>
     </hal>
-    <hal format="hidl" optional="true">
-        <name>hardware.google.light</name>
-        <version>1.0</version>
-        <interface>
-            <name>ILight</name>
-            <instance>default</instance>
-        </interface>
-      </hal>
     <hal format="aidl" optional="true">
         <name>vendor.google.wifi_ext</name>
         <version>1-3</version>
diff --git a/manifest.xml b/manifest.xml
index 482c16e5..1142dbb6 100644
--- a/manifest.xml
+++ b/manifest.xml
@@ -103,24 +103,6 @@
         <transport>hwbinder</transport>
         <fqname>@4.0::IKeymasterDevice/default</fqname>
     </hal>
-    <hal format="hidl">
-        <name>android.hardware.light</name>
-        <transport>hwbinder</transport>
-        <version>2.0</version>
-        <interface>
-            <name>ILight</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
-    <hal format="hidl">
-        <name>hardware.google.light</name>
-        <transport>hwbinder</transport>
-        <version>1.0</version>
-        <interface>
-            <name>ILight</name>
-            <instance>default</instance>
-        </interface>
-    </hal>
     <hal format="hidl">
         <name>android.hardware.media.c2</name>
         <transport>hwbinder</transport>
diff --git a/sepolicy/vendor/qcom/common/file_contexts b/sepolicy/vendor/qcom/common/file_contexts
index b87e44e9..db236ad9 100644
--- a/sepolicy/vendor/qcom/common/file_contexts
+++ b/sepolicy/vendor/qcom/common/file_contexts
@@ -154,7 +154,7 @@
 /vendor/bin/hw/hardware\.google\.media\.c2@1\.0-service-software u:object_r:mediacodec_exec:s0
 /vendor/bin/hw/vendor\.qti\.hardware\.display\.allocator@1\.0-service   u:object_r:hal_graphics_allocator_default_exec:s0
 /vendor/bin/hw/vendor\.google\.wireless_charger@1\.0-service-vendor       u:object_r:hal_wlc_exec:s0
-/vendor/bin/hw/hardware\.google\.light@1\.0-service                    u:object_r:hal_light_default_exec:s0
+/vendor/bin/hw/android\.hardware\.light-service\.lineage                 u:object_r:hal_light_default_exec:s0
 /vendor/bin/hw/vendor\.qti.hardware\.memtrack-service   u:object_r:hal_memtrack_default_exec:s0
 /vendor/bin/hw/android\.hardware\.boot-service\.qti     u:object_r:hal_bootctl_default_exec:s0
 
diff --git a/sepolicy/vendor/qcom/common/hwservice_contexts b/sepolicy/vendor/qcom/common/hwservice_contexts
index 2e9da598..abd7da47 100644
--- a/sepolicy/vendor/qcom/common/hwservice_contexts
+++ b/sepolicy/vendor/qcom/common/hwservice_contexts
@@ -23,4 +23,3 @@ vendor.nxp.nxpnfc::INxpNfc                                      u:object_r:nxpnf
 vendor.nxp.nxpese::INxpEse                                      u:object_r:nxpese_hwservice:s0
 vendor.google_paintbox::IManagerService                         u:object_r:hal_paintbox_hwservice:s0
 vendor.google.wireless_charger::IWirelessCharger                u:object_r:hal_wlc_hwservice:s0
-hardware.google.light::ILight                                   u:object_r:hal_light_hwservice:s0
